src/components/tile/tile.js

```diff
@@ -0,0 +1,88 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventMatches from '../../globals/js/misc/event-matches';
+
+class Tile extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * Tile.
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The element working as an Tile.
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.tileType = this.element.dataset.tile;
+    this._hookActions(this._getClass(this.tileType));
+  }
+
+  _getClass = (type) => {
+    const typeObj = {
+      expandable: this.options.classExpandedTile,
+      clickable: this.options.classClickableTile,
+      selectable: this.options.classSelectableTile,
+    };
+    return typeObj[type];
+  };
+
+  _hookActions = (tileClass) => {
+    const isExpandable = this.tileType === 'expandable';
+    if (isExpandable) {
+      const aboveTheFold = this.element.querySelector(this.options.selectorAboveTheFold);
+      if (aboveTheFold) {
+        this.tileHeight = this.element.getBoundingClientRect().height;
```

Comment:
this is more a readability thing, but I like declaring all the global variables up in the constructor and commenting them there, makes it easier to track when they get reused elsewhere; i.e.,

constructor {
   this.atfHeight = 0 // tracks above the fold height
}